### PR TITLE
Add metric-specific labels in googlecloudmonitoringreceiver component

### PR DESCRIPTION
**Pull request by abhishek-at-cloudwerx**
#### Description
Add metric-specific labels inside the googlecloudmonitoringreceiver component

#### Link to tracking issue
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35711

### CODE REVIEW

File: receiver/googlecloudmonitoringreceiver/receiver.go [hunk 1]
```diff
@@ -285,6 +285,13 @@ func (mr *monitoringReceiver) convertGCPTimeSeriesToMetrics(metrics pmetric.Metr
 			}
 		}
 
+		// Add metric-specific labels if they are present
+		if len(timeSeries.Metric.Labels) > 0 {
+			for k, v := range timeSeries.Metric.Labels {
```
Comment (dashpole) [inline receiver/googlecloudmonitoringreceiver/receiver.go:290]: Consider using functions to properly handle nil values
```suggestion
			for k, v := range timeSeries.GetMetric().GetLabels() {
```

**Review by dashpole — COMMENTED** [review]
please add unit tests

File: receiver/googlecloudmonitoringreceiver/receiver.go (hunk 1 above)
Comment (abhishek-at-cloudwerx) [reply receiver/googlecloudmonitoringreceiver/receiver.go]: Updated

**Comment by abhishek-at-cloudwerx** [comment]
> please add unit tests

Will create separate PR for it

**Comment by dashpole** [comment]
> Will create separate PR for it

Please add it to this PR.  Code and tests should be submitted together.

**Comment by dashpole** [comment]
I've been promised that tests are coming in a separate PR.

**Review by dashpole — APPROVED** [review]

**Pull request merged**